### PR TITLE
chore : Add relevant files to `paths-ignore`

### PR DESCRIPTION
**Pull request by epicadk**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

trivial

**Description**
Clear and concise code change description. 

Skip the workflow if changes are made only to .md files ( like `readme.md`) 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?
 
We could add the `license.txt` file as well but spotless check might fail if changes are made to the file so I've decided to include it. 

**Type**
Choose one: (Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,10 +19,14 @@ name: CI
 on:
   push:
     branches: [ master ]
+    paths-ignore: 
+      - '*.md'
```
Comment (jingtang10) [inline .github/workflows/ci.yml:23]:
```suggestion
      - '**.md'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,10 +19,14 @@ name: CI
...
     tags:
       run_ci*
   pull_request:
     branches: [ master ]
+    paths-ignore: 
+      - '*.md'
```
Comment (jingtang10) [inline .github/workflows/ci.yml:29]:
```suggestion
      - '**.md'
```

**Review by jingtang10 — CHANGES_REQUESTED** [review]
Thanks for this! Great idea using `paths-ignore`.

Please do modify the paths though to use `**` so that it covers subdirectories.

**Comment by epicadk** [comment]
> Thanks for this! Great idea using `paths-ignore`.
> 
> Please do modify the paths though to use `**` so that it covers subdirectories.

Yup sorry I wasn't aware of the difference between `*` and `**`. I've made the changes.

**Comment by epicadk** [comment]
I also added `CODEOWNERS` and `LICENSE` because they wouldn't really affect the outcome of the CI

**Comment by jingtang10** [comment]
Great! Thanks for this!

**Review by jingtang10 — APPROVED** [review]

**Comment by jingtang10** [comment]
This PR is causing some issues. If a PR is raised with changes only to the ignored files, no CI workflow will be triggered. But because we have a branch rule for `master` to ensure only PRs with passing tests can be merged, the PR cannot actually be merged.

I'm not sure this PR is still the right thing to do given this. Either we revert this PR, or if we can find a way for GitHub to not enforce the testing rule for PRs with only documentation change.

**Comment by jingtang10** [comment]
@deepankarb had this problem for #720

**Comment by epicadk** [comment]
I had a look at some of the actions repos and even they have removed the path filter due to a similar issue. (lastest commit on https://github.com/actions/upload-artifact). So I think removing the filter is the way to go.

The logic behind this change was to avoid running the tests because the code wouldn't have changed. So if we do find a way to implement this properly we could revisit this later.

**Comment by deepankarb** [comment]
Thanks @epicadk @jingtang10 The worst case scenario is when there are lots of small changes in documentation in multiple PRs (single PR updation will cancel any subsequent runs AFAIK so not an issue). Such a scenario is IMO rare. I think it's better to remove the `paths-ignore`.

**Pull request merged**